src/components/RepoSearchForm.tsx

```diff
@@ -8,17 +8,29 @@ import { GithubRepositoryItem } from '../modules/github/types';
 import './RepoSearchForm.css';
 import Search from 'antd/lib/input/Search';
 
+const INITIAL_PAGE_SIZE = 10;
+const MAX_PAGES_AMOUNT = 1000;
+
 export const GithubSearchRepositories: React.FC = () => {
     const dispatch: any = useDispatch();
+    const {list, isLoading, totalRepositoriesCount} = useSelector((state: StoreState) => state.gitHubRepositories);
     const [searchName, setSearchName] = useState('');
-    const { list, isLoading } = useSelector((state: StoreState) => state.gitHubRepositories);
+    const [pageSize, setPageSize] = useState<number>(INITIAL_PAGE_SIZE);
 
     const onChangeRepositoryTitle = (e: ChangeEvent<HTMLInputElement>) => setSearchName(e.target.value);
-
-    const onClick = () => {
-        dispatch(searchGithubRepositories(searchName));
+    const totalPagesAmount = totalRepositoriesCount <= MAX_PAGES_AMOUNT && totalRepositoriesCount;
```

Comment:
Why do we need this? In particular, why do we need `MAX_PAGES_AMOUNT`?

Reply:
The request from the server is limited to 1000 pages, so we need to limit the number of pages displayed in order to avoid an error.